lua/nvim-tree/explorer/reload.lua

```diff
@@ -65,6 +65,7 @@ function M.reload(node, status)
       local n = nodes_by_path[abs]
       if n then
         n.executable = builders.is_executable(abs, n.extension or "")
+        n.fs_stat = uv.fs_stat(abs)
```

Comment:
This extra stat is expensive and unnecessary.

Please retain a value from the start of the loop:
```lua
    local stat = uv.fs_stat(abs)
    t = t or (stat or {}).type
```

Reply:
Looks like the first stat is rarely necessary since `uv.fs_scandir_next` returns the type for most file systems I tested with (ext3, zfs, vfat) _except_ reiserfs 💀.

Refactored the code to call `stat` only when necessary.

Reply:
Nice work.